crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -471,7 +472,15 @@ impl WasmCtx {
 fn format_files(generated_files: Vec<PathBuf>, lang: Language) -> anyhow::Result<()> {
     match lang {
         Language::Rust => {
-            cmd!("rustup", "component", "add", "rustfmt").run()?;
+            if !has_rust_fmt() {
+                if has_rust_up() {
+                    if let Err(err) = cmd!("rustup", "component", "add", "rustfmt").run() {
+                        println!("Warning: Failed to install rustfmt: {err}");
+                    }
+                } else {
+                    return Err(anyhow::anyhow!("rustfmt is not installed. Please install it."));
```

Comment:
it's a bit strange to me that if we _fail_ to install `rustfmt` above, we just print a warning, but if we don't have `rustup` at all then we return an error. Should we print a warning in this case too, or should we return an `Err` above?